pkg/chunk/cache/cache_gen_test.go

```diff
@@ -0,0 +1,39 @@
+package cache
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/net/context"
+)
+
+func TestCacheGenNumCacheKeysPrefix(t *testing.T) {
+	keys := []string{"foo", "bar", "baz"}
+
+	for _, tc := range []struct {
+		name   string
+		prefix string
+	}{
+		{
+			name: "empty-prefix",
+		},
+		{
+			name:   "with-prefix",
+			prefix: "prefix",
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx := InjectCacheGenNumber(context.Background(), tc.prefix)
+
+			prefixedKeys := addCacheGenNumToCacheKeys(ctx, keys)
+			for i, key := range prefixedKeys {
+				require.Equal(t, tc.prefix+keys[i], key)
+			}
+
```

Comment:
Also worth checking if the length of the `keys`, `prefixedKeys` and `unprefixedKeys` is the same.